src/core/domain.py

```diff
@@ -86,3 +86,21 @@ def tls_ca_chain(self) -> List[str] | None:
     def log_dir(self) -> str:
         """Return the full path to the object."""
         return f"s3a://{self.bucket}/{self.path}"
+
+
+class PushGatewayInfo(StateBase):
+    """Class representing thr endpoints to connect to the prometheus PushGateway."""
+
+    def __init__(self, relation: Relation, component: Application):
+        super().__init__(relation, component)
+
+    @property
+    def endpoint(self) -> str | None:
+        """Return endpoint of the Prometheus PushGateway."""
+        raw_data = self.relation_data.get("push-endpoint", None)
+        if raw_data:
+            data = json.loads(raw_data)
+            if "url" in data:
+                url = data["url"]
```

Comment:
**thought** that's a pity that the library does not provide an abstraction over the data relation, to prevent people from referring to the low-level fields. i thought that the COS team would generally do that. anyhow, given the current status of the lib, I don't see any other way